scripts/biomedical/VirtualMetabolicHuman/format_reaction.py

```diff
@@ -0,0 +1,221 @@
+    df_metabolite = pd.read_csv(file_metabolite)
+    
+    df_rxn["subsystem"]  = df_rxn["subsystem"].str.lower()
+  
+    
+    # Format the subsystem field
+    for i in df_rxn.index:
+        l = df_rxn.loc[i,'subsystem']
+        l = l.replace(',','')
+        df_rxn.loc[i,'subsystem'] = l
+        #p = n.replace(",","")
+        #m = p.replace(" ","_")
+    
+    for i in df_rxn.index:
+        l = df_rxn.loc[i,'subsystem']
+        l = l.replace(' ','_')
+        df_rxn.loc[i,'subsystem'] = l
+        #p = n.replace(",","")
+        #m = p.replace(" ","_")
+    list_match = ['0']*len(df_rxn['abbreviation'])
+    list_dcids = ['0']*len(df_rxn['abbreviation'])
+  
+    # Perform a match between metabolites and reactions
+    # Using abbreviations for both
+    for p in range(len(df_rxn['abbreviation'])):
+        for q in df_metabolite.index:
+            test_list = Convert(df_rxn.loc[p,'abbreviation'])
+            for i in range(len(test_list)):
+                test_list[i] = test_list[i].lower()
+            sub_list = Convert(df_metabolite.loc[q,"abbreviation"])
+            if(is_subset(test_list, sub_list)):
+                if(list_match[p] != '0'):
+                    if(len(list_match[p]) < len(sub_list)):
+                        list_match[p] = sub_list
+                        list_dcids[p] = df_metabolite.loc[q,"Id"]
+                else:
+                    list_dcids[p] = sub_list
+                    list_dcids[p] = df_metabolite.loc[q,"Id"]
+
+    df_rxn['MetaboliteMatch'] = list_dcids
+    
+    
+    #Map the metanetx to humanGemIDs using human1D data
+
+    # create new db with compartments from human gem id (2)
+    #df_map_new = pd.DataFrame()
+    df_map_new = df_map[["id", "reactant_compartment", "product_compartment"]].copy()
+    l1 = df_map_new['id'].str[2:]
+    df_map_new = df_map_new.drop('id', 1)
+    df_map_new['humanGEMID'] = l1
+
+    hgem_col = []
+    for i in df_rxn["metanetx"].values:
+        hgem_col.append(df_map[df_map["metanetx"] == i]["id"].values)
+    df_rxn.insert(value = hgem_col, column = "humanGEMID", loc = 0)
+    df_rxn = df_rxn.explode(column = "humanGEMID")
+    df_rxn["humanGEMID"] = np.where(pd.isnull(df_rxn['humanGEMID']),df_rxn['humanGEMID'] \
+                                    ,df_rxn['humanGEMID'].str[2:])
+    df_rxn["Id"] = np.where(pd.isnull(df_rxn['humanGEMID']),df_rxn['humanGEMID'] \
+                                   ,"bio/" +df_rxn['humanGEMID'].astype(str))
+
+
+    #merge two db with compartment info from human gemid (4)
+    df = pd.merge(df_rxn, df_map_new, on='humanGEMID')
+
+    db = df
+    db_dup = db
+    #db_dup = db.sample(n = 400)
+    # dict to map compartment info b/w gemid and formula (vhm -> human 1d)
+    matchdict = {
+    "e": "s",
+    "x": "p",
+    "i": "c_i",
+    "c": "c",
+    "s": "s",
+    "l": "l",
+    "r": "r",
+    "m": "m",
+    "g": "g",
+    "n": "n"
+        
+    }
+    chars_to_check = '='
+    for i, row in db_dup.iterrows():
```

Comment:
consider cleaning up this loop and the next one: remove commented out print statements, add more comments (e.g., for the long condition in line 140)

Reply:
Done ! Removed the extra spaces and print statements.